### PR TITLE
Improving Models page content maintainability and UX

### PR DESCRIPTION
**Pull request by aaronsarnat**
**This PR:**

- Fixes [Simplify maintenance of Models page content by migrating to a data-driven Jekyll template (#65)](https://github.com/allenai/allennlp-website/issues/65)
- Fixes [Models page side-nav links should auto-select based on scroll position (#64)](https://github.com/allenai/allennlp-website/issues/64)
- Substantially reduces boilerplate for model content management
- Introduces `.yml` data files for models and gallery items
- Auto-generates Models page side-nav
- Gives us a net savings of 247 LOC

Aside from the single UX enhancement that now shows the user's place in the table of contents (see screenshot), the output should look identical to what is currently on master.

image

### CODE REVIEW

File: _data/gallery.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+- title:    "Deep enhanced representation for implicit discourse relation recognition‏"
```
Comment (aaronsarnat) [inline _data/gallery.yml]: @schmmd FYI this is what Gallery content data looks like now

File: _data/models.yml [hunk 2]
```diff
@@ -0,0 +1,203 @@
+- title: "Machine Comprehension"
```
Comment (aaronsarnat) [inline _data/models.yml]: @schmmd FYI this is what Model content data looks like now

File: models.html [hunk 3]
```diff
@@ -24,687 +25,35 @@ <h1>Models</h1>
         </div>
       </div>
 
-      <div class="col-layout constrained col-layout--hidden-mobile-sidenav">
+      <div id="auto-nav" class="col-layout constrained col-layout--hidden-mobile-sidenav">
         <div class="pure-g">
           <div class="col-layout__nav pure-u-1 pure-u-lg-1-4">
             <nav>
-              <a href="#top" title="Scroll to top of page"><h4>Our Models</h4></a>
```
Comment (aaronsarnat) [inline models.html]: Side-nav links are now auto-generated

File: models.html [hunk 4]
```diff
@@ -24,687 +25,35 @@ <h1>Models</h1>
...
                   <p class="t-sm">We provide the best models we trained.  In other words, we have trained our models multiple times and selected the best result.  Please keep in mind that if you retrain any of the following models, you are likely to end up with a slightly worse evaluation.</p>
                 </div>
               </div>
             </div>
-            <!-- Machine Comprehension -->
```
Comment (aaronsarnat) [inline models.html]: All the model boilerplate is gone, and now this page is pretty small. It's pretty much just the static content that isn't a model or a user gallery item.

**Comment by schmmd** [comment]
Looks great!

**Pull request merged**